lib/reek/context/class_context.rb

```diff
@@ -0,0 +1,11 @@
+require_relative 'module_context'
+
+module Reek
+  module Context
+    #
+    # A context wrapper for any class found in a syntax tree.
+    #
+    class ClassContext < ModuleContext
```

Comment:
>> If this is the case doesn't it make more sense to move #method_calls here?

I'd leave it here since the method's purpose is not tied to the SmellDetector.

>> (But perhaps this smell should also check modules?)

But you don't know if a private module method is called or not since it might get mixed into a class.....or not. And we don't know that. We *would* know it if Reek would slurp in all the sources and build up a call / class / whatever registry for later usage but with our current "one file == one examiner" paradigm this wouldn't work.